File: Data/Entities/Product.cs
using System.Collections.Generic;

namespace WebStore.Data.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public Category Category { get; set; }
        public int CategoryId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Price { get; set; }
        public string Currency { get; set; }
        public int Amount { get; set; }
        public List<Opinion> Opinions { get; set; }
        public double AverageRating { get; set; }
    }
}

File: WebStoreFrontend/src/app/data/product.ts
import { Opinion }from  "src/app/data/opinion";
import { Category } from "src/app/data/category"

export class Product {
  id: number
  category: Category
  name: String
  description: String
  price: number
  currency: String
  amount: number
  averageRating: number
  opinions: Opinion[]
}
